src/common/numeric.h

```diff
@@ -32,9 +32,7 @@ void RunLengthEncode(Iter begin, Iter end, std::vector<Idx>* p_out) {
       out.push_back(i);
```

Comment:
According to the above for loop
`out.push_back(n);`
This line also redundant?
Or is there something wrong with my understanding?
@trivialfis

Reply:
A quick counter-example comes to mind, what if n==0?

Reply:
> A quick counter-example comes to mind, what if n==0?

Thanks for your answer！